extensions/chathistory-3.3.md

```diff
@@ -0,0 +1,79 @@
+
+The `chathistory` extension adds an optional `draft/msgid` to the `chathistory` batch reply.
+
+Client side support for the [`batch`][batch], [`server-time`][server-time], and [`draft/labeled-response`][draft/labeled-response] capabilities is required. Support for the [`draft/msgid`][draft/msgid] capability is optional but recommended.
+
+When a client with the above mentioned capabilities requests `chathistory` content from the server (using the `CHATHISTORY` command outlined below), the server should return to the client a single batch containing a number of desired raw IRC lines equal to the `message_count` parameter specified, ending directly before the given timestamp or with the message directly proceeding the one with the specified `draft/msgid`. The raw IRC lines should be formatted and returned to the client as they were originally, with the addition of the above capability tags.
+
+The returned `batch` must not be blank. If no content exists to return, the appopriate error message should be sent instead.
+
+Both the `message_count` and `max_message_count` must be integers greater than 0. The client should not request a `message_count` greater than the `max_message_count` parameter sent in the command. If the `message_count` exceeds the `max_message_count`, server should return a number of lines equal to the `max_message_count`.
+
+The `server-time` should be the time at which the message was originally sent and the `batch id` a unique ID to the entire batch. The `draft/label` should be a unique ID used to identify the `chathistory` request and any replies. The optional `draft/msgid`, if included, should be the `draft/msgid` originally sent with the message.
+
+### `CHATHISTORY` Command
+Chathistory content can be requested by the client to the server by sending the `CHATHISTORY` command to the server. A `batch` must be returned by the server. If no content exists to return, an empty batch should be returned to avoid the client waiting for a reply. Command support is sent to the client as the RPL_ISUPPORT 005 numeric `:irc.host 005 nick CHATHISTORY=max_message_count :are supported by this server`
+
+#### Format
+The `chathistory` content can requested using timestamps:
+
+    @draft/label=ID CHATHISTORY target timestamp=YYYY-MM-DDThh:mm:ss.sssZ message_count
+
+Alternatively, content can be requested using a `draft/msgid`:
+
+    @draft/label=ID CHATHISTORY target draft/msgid=ID message_count
+
+If no message_count is known, `*` should be used to specify the default value.
+
+#### Errors
+If the server receives an improperly formatted `CHATHISTORY` command, the `CMD_INVALID` error code should be returned.
+
+If no `chathistory` exists to return, the server should return the appropriate error code. `ACCESS_DENIED` should be sent if the user requests content they do not have permission to view. `NOT_FOUND` should be sent if no content matching the request is found and no `ACCESS_DENIED` error exists.
```

Comment:
Addressed in 	78b39f1 - the `max_message_count` should be returned an `WARN MAX_MESSAGE_COUNT_EXCEEDED` returned to let the client know the value was too high.